master.cfg

```diff
@@ -313,33 +313,57 @@ c['schedulers'].append(
+ubiquity_base.addStep(steps.ShellCommand(command=["/usr/bin/curl", "-H", "x-amz-acl: public-read", upload_url, "--upload-file", compressed_image_file]))
+ubiquity_base.addStep(steps.ShellCommand(command=["/usr/bin/sha256sum", compressed_image_file]))
+ubiquity_base.addStep(steps.FileUpload(name="Upload Image", workersrc=compressed_image_file, masterdest=compressed_image_master_dest))
 
 c['builders'].append(
-    util.BuilderConfig(name="pi-lxde-focal-testing-image",
+    util.BuilderConfig(name="pi-base-focal-testing-image",
       workernames=["boron"],
-      factory=ubiquity_lxde))
+      factory=ubiquity_base))
 
 # Weekly testing image built at midnight UTC every monday
-c['schedulers'].append(
-    schedulers.Nightly(name='weekly-focal-testing-image',
-                       builderNames=['pi-lxde-focal-testing-image'],
-                       dayOfWeek=0, hour=0, minute=1))
+# c['schedulers'].append(
+#     schedulers.Nightly(name='weekly-focal-testing-image',
+#                        builderNames=['pi-base-focal-testing-image'],
+#                        dayOfWeek=0, hour=0, minute=1))
+
+
+
+ubiquity_gdb = util.BuildFactory()
+ubiquity_gdb.addStep(steps.Git(repourl='https://'+creds.buildbot_github_token+'@github.com/UbiquityRobotics/pi_image2.git', mode='incremental'))
+# first need to install debootstrap and pychroot - TODO do this in the buldbot image template so we dont have to do it everytime here
+ubiquity_gdb.addStep(steps.ShellCommand(command= ["sudo", "apt-get", "install", "python3-pip", "debootstrap", "-y"], description="Install pip and chroot", name="Install pip and chroot", timeout=4000))
+ubiquity_gdb.addStep(steps.ShellCommand(command= ["sudo", "pip3", "install", "pychroot"], description="Install chroot", name="Install chroot", timeout=4000))
+ubiquity_gdb.addStep(steps.ShellCommand(command= ["sudo", "python3", "build_image.py", "--config_path", "focal-dev-build-settings.yaml"], description="Building", name="Run Build Script", timeout=4000))
+ubiquity_gdb.addStep(steps.SetPropertyFromCommand(command="cat latest_image", property="latest_image_path", haltOnFailure=True))
+compressed_image_file = util.Interpolate('%s.xz', util.Property('latest_image_path'))
+image_name = util.Transform(os.path.basename, util.Property('latest_image_path'))
+compressed_image_name = util.Interpolate('%s.xz', image_name)
+compressed_image_master_dest = util.Interpolate('/buildwork/%s.xz', image_name)
+
+upload_url = util.Transform(generate_image_upload_url, compressed_image_name)
+ubiquity_gdb.addStep(steps.ShellCommand(command=["/usr/bin/curl", "-H", "x-amz-acl: public-read", upload_url, "--upload-file", compressed_image_file]))
+ubiquity_gdb.addStep(steps.ShellCommand(command=["/usr/bin/sha256sum", compressed_image_file]))
+ubiquity_gdb.addStep(steps.FileUpload(name="Upload Image", workersrc=compressed_image_file, masterdest=compressed_image_master_dest))
+
+c['builders'].append(
+    util.BuilderConfig(name="pi-gdm3-focal-dev-image",
```

Comment:
This is fine for now, but we really should get this build to be on top of the base image generated by the previous step.

Reply:
yes, eventually